### PR TITLE
Public CI

### LINKED ISSUE #8: Public CI
Add Travis or CircleCI build anyone can see, before making the repo public.

### PR DESCRIPTION
**Pull request by kim**
Fixes #8

### CODE REVIEW

**Comment by kim** [comment]
Provides a publicly-visible CI (travis), and buids using `cabal-install` instead of `stack` (same reasoning as oscoin/ipfs#30).

Additionally, Goog Clown Build is killed off entirely, as it does not provide any additional value.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+install:
+  - cabal v2-update
+  - cabal v2-install -j --overwrite-policy=always stylish-haskell hlint
+
+script:
+  - .ci/check-haskell-format.sh
+  - hlint .
+  - cabal v2-configure --enable-tests
+  # wtflol:
+  - 'echo -e "package gossip\n  ghc-options: -Werror" >> cabal.project.local'
```
Comment (geigerzaehler) [inline .travis.yml]: I’d be interested why we need to do this. Could you add a comment?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  - cat cabal.project.local
```
Comment (geigerzaehler) [inline .travis.yml]: Do we still need this?

Comment (kim) [reply .travis.yml]: Yes. I'm at a point where I don't trust the cabal folks anymore to not silently introduce some breakage to `configure`, so I'd rather want to know what the configuration is.

**Pull request merged**